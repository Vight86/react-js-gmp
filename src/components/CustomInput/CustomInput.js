import classNames from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';
import s from './CustomInput.scss';

const CustomInput = ({
  type, name, value, handleChange, label, placeholder, theme, errorMessage,
}) => {
  const changeType = (e) => {
    e.target.type = 'date';
  };

  const classes = classNames({
    [s[theme]]: true,
    [s.primary_withLabel]: !!label,
  });

  const isDate = type === 'date';

  return (
    <label className={classes}>
      {label && label}
      {isDate
        ? (
          <input
            name={name}
            placeholder="Select date"
            type="text"
            onFocus={changeType}
          />
        )
        : (
          <input
            name={name}
            type={type}
            value={value}
            placeholder={placeholder}
            onChange={handleChange}
          />
        )}
      {errorMessage && <span className={s.error}>{errorMessage}</span>}
    </label>

  );
};

CustomInput.propTypes = {
  name: PropTypes.string,
  label: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  handleChange: PropTypes.func,
  type: PropTypes.oneOf(['text', 'password', 'email', 'url', 'date', 'number']).isRequired,
  placeholder: PropTypes.string,
  theme: PropTypes.string.isRequired,
  errorMessage: PropTypes.string,
};

CustomInput.defaultProps = {
  name: '',
  label: '',
  placeholder: '',
  errorMessage: '',
  handleChange: () => {},
};

export default CustomInput;
