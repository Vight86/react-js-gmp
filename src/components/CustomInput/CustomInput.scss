@use '../../scss/variables';

.primary {
  width: 100%;
  font-family: inherit;
  font-style: normal;
  font-weight: 600;
  font-size: 1.6em;
  color: variables.$color-primary;
  letter-spacing: 0.888889px;
  text-transform: uppercase;

  input {
    width: 100%;
    display: block;
    border-radius: 0.32em;
    font-size: 1.25em;
    height: 3.1em;
    font-family: inherit;
    font-weight: 400;
    color: variables.$color-fivefold;
    padding: 0 1em;
    transition: all .3s;
    border: 0.125em solid transparent;
    background-color: transparentize(variables.$color-secondary, 0.4);
    box-sizing: border-box;

    &:focus {
      border: 0.125em solid variables.$color-primary;
    }

     &[type="date"]  {
        &::-webkit-calendar-picker-indicator {
          margin-top: 0.8em;
          background: url("datePickerIcon.svg") no-repeat
        }

        &::-webkit-datetime-edit {
          line-height: 2.7em;
        }
      }
  }

  &_withLabel {
    position: relative;
    input {
      margin-top: 0.7em;
    }
  }
}

.error {
  position: absolute;
  font-family: inherit;
  font-style: normal;
  font-weight: 400;
  font-size: 0.85em;
  color: variables.$color-primary;
  opacity: 0.8;
  bottom: -1.7em;
  left: 0;
}
