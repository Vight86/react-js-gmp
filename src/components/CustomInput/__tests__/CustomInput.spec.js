import { render } from '@testing-library/react';
import React from 'react';
import CustomInput from '../CustomInput';

describe('CustomInput', () => {
  const setup = () => render(<CustomInput
    theme="primary"
    type="text"
  />);

  test('render primary theme correctly', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
