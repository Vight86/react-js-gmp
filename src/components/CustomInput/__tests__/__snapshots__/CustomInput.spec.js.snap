// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput render primary theme correctly 1`] = `
<div>
  <label
    class="primary"
  >
    
    <input
      name=""
      placeholder=""
      type="text"
      value=""
    />
    
  </label>
</div>
`;
